log4j-docgen/src/main/java/org/apache/logging/log4j/docgen/internal/DefaultSchemaGenerator.java

```diff
@@ -0,0 +1,351 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.logging.log4j.docgen.internal;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.TreeSet;
+import javax.xml.XMLConstants;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamWriter;
+import org.apache.logging.log4j.docgen.SchemaGenerator;
+import org.apache.logging.log4j.docgen.model.Description;
+import org.apache.logging.log4j.docgen.model.EnumType;
+import org.apache.logging.log4j.docgen.model.EnumValue;
+import org.apache.logging.log4j.docgen.model.PluginAttribute;
+import org.apache.logging.log4j.docgen.model.PluginBundle;
+import org.apache.logging.log4j.docgen.model.PluginElement;
+import org.apache.logging.log4j.docgen.model.PluginEntry;
+import org.apache.logging.log4j.docgen.model.io.stax.PluginBundleStaxReader;
+
+public class DefaultSchemaGenerator implements SchemaGenerator {
```

Comment:
Do we expect `SchemaGenerator` to be extensible? How? In the first release? If not, I would prefer keeping it simple and moving the entire logic to `SchemaGenerator` instead.

Reply:
The idea behind this was to follow Maven's practice.
They usually create an interface for each service and inject it in the plugin.

Reply:
How do they inject it? If we are gonna statically link the implementation class at the injection site anyway, what is the advantage of this approach?

Reply:
The recent way is to use [Sisu](https://eclipse.dev/sisu/), which requires me to add some `@Named` and `@Singleton` annotations.